### ISSUE TITLE
Automatically configure CloudTrail when creating cloud account

### ISSUE THREAD

**Issue by infinitewarp** [cloudigrade/cloudigrade#214]
As a user of cloudigrade, I want my AWS CloudTrail to send logs to cloudigrade with minimum necessary permissions (and an explanation of them) so that cloudigrade can monitor and track my instance usage.

Problems:
- We need to automagically set up our CloudTrail to work with new customers.
- We need to ask different/more permissions of our customers' AWS IAM. Figure out _really_ what the policy needs at minimum now that we are including CloudTrail.
- Since IAM adding is still manual, we need to update any documents (and UI bubbles?) to explain to someone how to do this.

## Acceptance Criteria 

- [x] Verify that documentation with step-by-step AWS account setup instructions have been updated to reflect the addition of CloudTrail.
- [x] Verify that removing any further permissions will break functionality. (This means we have produced the minimum set of permissions.)
- [x] Verify that during account registration, cloudigrade configures the customer's account for CloudTrail.
- [x] Verify that the customer _user_ cannot write into our S3, _but_ CloudTrail can write on their behalf.
- [x] Verify integration test issue exists (see https://github.com/cloudigrade/integrade/issues/36).

## Assumptions and Questions

- Customer's Trail needs to be configured with: 
  - capture only WRITE events (not READ events)
  - send logs to a known existing S3 bucket that belongs to the cloudigrade "main" AWS account
  - The "main" AWS account S3 bucket should notify an existing queue in the same amount when something is logged to the bucket
- As part of this story, define for other cloudigrade developers how to set up the S3 bucket and SQS queue, and the thing in AWS that automagically sends notifications to the SQS queue in our account.

**Comment by jlprevatt** [comment]
RE: "Since IAM adding is still manual, we need a document *(do we write this?)* to explain to someone how to do this."

Yes, this needs to be captured in some kind of written form so that that writeup can be verified by dev team testing and QE testing. When it is rock-solid (or as close as we can get) then it can be examined/reworked for whatever the documentation solution for Cloud Metering is going to be.

TLDR version: This is not information that I would obtain via a verbal interview.

**Comment by abaiken** [comment]
Demo: [Demo: https://drive.google.com/file/d/17N20InLCL0hZeJ0B-bgIT0zj6rpuxJXj/view?usp=sharing](Demo: https://drive.google.com/file/d/17N20InLCL0hZeJ0B-bgIT0zj6rpuxJXj/view?usp=sharing)

**Comment by infinitewarp** [comment]
Demo looks good to me and appears to ✔️ most of the ACs! Please check in with our friendly neighborhood QEs to discuss potential integration tests.

**Comment by elyezer** [comment]
Now that we have the integrade issue we are good to move this to closed.

**Comment by kdelee** [comment]
@jlprevatt @infinitewarp @bclarhk as I was working on testing this functionality, I noticed this little banner on my AWS console:
```
Trails
Deliver logs to an Amazon S3 bucket.
 CloudTrail events can be processed by one trail for free.
 There is a charge for processing events with additional trails.
 For more information, see AWS CloudTrail Pricing.
```

Seems like we need to notify users about this because if they are already using CloudTrail, maybe it would cost money? Maybe this information is already captured somewhere? Just thought I'd make note of this.
